core/src/test/java/test/github2360/Issue2360Test.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2010-2021 Evergage, Inc.
+ * All rights reserved.
+ */
+
+package test.github2360;
+
+import org.testng.IMethodInstance;
+import org.testng.IMethodInterceptor;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * A unit test for https://github.com/cbeust/testng/issues/2360.
+ */
+@Test(description = "GITHUB-2360")
+public class Issue2360Test extends SimpleBaseTest {
+
+    public void testGroovyInternalMethodsAreSkipped() {
```

Comment:
Can we add this as a test method into the existing test class `test.groovy.GroovyTest` so that all the groovy related tests are in the same java class? You could add a description attribute which refers to the github issue

https://github.com/cbeust/testng/blob/master/core/src/test/resources/testng.xml#L686